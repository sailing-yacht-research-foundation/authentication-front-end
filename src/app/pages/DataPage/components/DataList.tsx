import { translations } from 'locales/translations';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { StyleConstants } from 'styles/StyleConstants';
import { DataDetails } from './DataDetails';
import i18next from 'i18next';
import { media } from 'styles/media';

const dataArray = [
    {
        title: i18next.t(translations.data_page.races_start_as_points),
        description: i18next.t(translations.data_page.an_animated_view_of_individual),
        screenshotUrl: '/static-htmls/A_race_starts/screenshot.png',
        htmlUrl: '/static-htmls/A_race_starts/race_start_locations.html',
    },
    {
        title: i18next.t(translations.data_page.race_clusters),
        description: i18next.t(translations.data_page.race_starts_grouped),
        screenshotUrl: '/static-htmls/B_race_clusters/screenshot.png',
        htmlUrl: '/static-htmls/B_race_clusters/race_clusters.html',
    },
    {
        title: i18next.t(translations.data_page.race_regions),
        description: i18next.t(translations.data_page.polygons_containing_races),
        screenshotUrl: '/static-htmls/C_regions/screenshot.png',
        htmlUrl: '/static-htmls/C_regions/sailing_regions.html',
    },
    {
        title: i18next.t(translations.data_page.race_region_boxes),
        description: i18next.t(translations.data_page.similar_to_racec_regions),
        screenshotUrl: '/static-htmls/D_region_boxes/screenshot.png',
        htmlUrl: '/static-htmls/D_region_boxes/boxes_around_regions.html',
    },
    {
        title: i18next.t(translations.data_page.yatch_club_density),
        description: i18next.t(translations.data_page.hexbin_map_to_visualize),
        screenshotUrl: '/static-htmls/E_yacht_clubs/screenshot.png',
        htmlUrl: '/static-htmls/E_yacht_clubs/yacht_clubs.html',
    }
];

export const DataList = () => {

    const [showDetail, setShowDetail] = React.useState<boolean>(false);

    const [selectedDataItem, setSelectedDataItem] = React.useState<any>({});

    const { t } = useTranslation();

    const renderDataItem = () => {
        return dataArray.map((data, index) => {
            return <ItemContainer key={index} onClick={e => renderDataDetail(data)}>
                <ItemScreenshot style={{ background: `url(${data.screenshotUrl})`, backgroundSize: 'cover', backgroundRepeat: 'no-repeat' }} />
                <ItemInfoContainer>
                    <ItemTitle>{data.title}</ItemTitle>
                    <ItemDescription>
                        {data.description}
                    </ItemDescription>
                </ItemInfoContainer>
            </ItemContainer>;
        });
    }

    const renderDataDetail = (data) => {
        setSelectedDataItem(data);
        setShowDetail(true);
    }

    const goBack = () => {
        setShowDetail(false);
    }

    return (
        <>
            {
                !showDetail ? (
                    < Wrapper >
                        <PageHeading>{t(translations.data_page.syrf_race_data)}</PageHeading>
                        {renderDataItem()}
                    </Wrapper >
                )
                    : (<DataDetails goBack={goBack} data={selectedDataItem} />)
            }
        </>
    )
}

const Wrapper = styled.div`
    padding: 10px 15px;
    margin-top: ${StyleConstants.NAV_BAR_HEIGHT};
`;

export const PageHeading = styled.h2`
    margin-top: 30px;
    margin-bottom: 30px;
`;

const ItemContainer = styled.div`
    width: 100%;
    display: flex;
    margin-bottom: 15px;
    align-items: center;

    :not(:last-child) {
        border-bottom: 1px solid #eee;
    }
`;

const ItemScreenshot = styled.div`
    width: 100px;
    height: 100px;
    margin-right: 30px;
    flex-shrink: 0;
`;

const ItemInfoContainer = styled.div`
    display:flex;
    flex-direction: column;
    padding-top: 15px;

    ${media.medium`
        padding-top: 0;
    `}
`;

const ItemTitle = styled.h4`
    color: #1890ff;
    cursor: pointer;
`;

const ItemDescription = styled.p`
    margin-top: 5px;
`;