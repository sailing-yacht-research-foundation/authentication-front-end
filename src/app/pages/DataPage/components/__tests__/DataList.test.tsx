import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { DataList } from '../DataList';

const shallowRenderer = createRenderer();

describe('<DataList />', () => {
  it('should render the deals page and matchs the snapshot', () => {
    shallowRenderer.render(<DataList />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});