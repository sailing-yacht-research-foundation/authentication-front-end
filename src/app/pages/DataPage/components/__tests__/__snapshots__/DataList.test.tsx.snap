// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataList /> should render the deals page and matchs the snapshot 1`] = `
<React.Fragment>
  <styled.div>
    <styled.h2>
      SYRF Race Data
    </styled.h2>
    <styled.div
      onClick={[Function]}
    >
      <styled.div
        style={
          Object {
            "background": "url(/static-htmls/A_race_starts/screenshot.png)",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
          }
        }
      />
      <styled.div>
        <styled.h4>
          Race Starts as Points
        </styled.h4>
        <styled.p>
          An animated view of individual races over the last few decades. Each dot is a unique race.
        </styled.p>
      </styled.div>
    </styled.div>
    <styled.div
      onClick={[Function]}
    >
      <styled.div
        style={
          Object {
            "background": "url(/static-htmls/B_race_clusters/screenshot.png)",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
          }
        }
      />
      <styled.div>
        <styled.h4>
          Race Clusters
        </styled.h4>
        <styled.p>
          Race starts grouped into clusters based on the scale of the map view.
        </styled.p>
      </styled.div>
    </styled.div>
    <styled.div
      onClick={[Function]}
    >
      <styled.div
        style={
          Object {
            "background": "url(/static-htmls/C_regions/screenshot.png)",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
          }
        }
      />
      <styled.div>
        <styled.h4>
          Race Regions
        </styled.h4>
        <styled.p>
          Polygons containing races for the last 20 years with a count of how many races occured in that region.
        </styled.p>
      </styled.div>
    </styled.div>
    <styled.div
      onClick={[Function]}
    >
      <styled.div
        style={
          Object {
            "background": "url(/static-htmls/D_region_boxes/screenshot.png)",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
          }
        }
      />
      <styled.div>
        <styled.h4>
          Race Region Boxes
        </styled.h4>
        <styled.p>
          Similar to race regions, except this is a view of nice rectangular boxes that contain each region. Overlapping boxes have been combined.
        </styled.p>
      </styled.div>
    </styled.div>
    <styled.div
      onClick={[Function]}
    >
      <styled.div
        style={
          Object {
            "background": "url(/static-htmls/E_yacht_clubs/screenshot.png)",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
          }
        }
      />
      <styled.div>
        <styled.h4>
          Yacht Club Density
        </styled.h4>
        <styled.p>
          3D Hexbin map to visualize the density of yacht clubs.
        </styled.p>
      </styled.div>
    </styled.div>
  </styled.div>
</React.Fragment>
`;
